oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/TokenHandler.java

```diff
@@ -147,16 +147,24 @@ public APIGatewayProxyResponseEvent tokenRequestHandler(
                             }
 
                             Map<String, String> requestBody = parseRequestBody(input.getBody());
-                            String clientID = requestBody.get("client_id");
-                            attachLogFieldToLogs(CLIENT_ID, clientID);
-                            addAnnotation("client_id", clientID);
                             addAnnotation("grant_type", requestBody.get("grant_type"));
 
+                            String clientID;
                             ClientRegistry client;
                             try {
+                                if (requestBody.get("client_id") != null) {
+                                    clientID = requestBody.get("client_id");
```

Comment:
we could remove this and retrieve it from the secure jwt every time?

Reply:
Yes to be in an another PR after seeing who's using it.